src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs

```diff
@@ -114,7 +114,12 @@ public void AddEntry<TData>(DebugDirectoryEntryType type, uint version, uint sta
             }
 
             // We allow NUL characters to allow for padding for backward compat purposes.
-            if (pdbPath.Length == 0 || pdbPath.IndexOf('\0') == 0)
+            if (pdbPath.Length == 0 ||
+#if NETCOREAPP
+                pdbPath.StartsWith('\0'))
+#else
+                pdbPath.Length > 0 && pdbPath[0] == '\0')
+#endif
```

Comment:
```suggestion
            if (pdbPath.Length == 0 ||
#if NETCOREAPP
                pdbPath.StartsWith('\0'))
#else
                pdbPath[0] == '\0')
#endif
```
BTW, is that a perf win, or loss? I would expect StartsWith(char) on a string known to be not empty would be slower. This should probably stay how it was?

Reply:
Looks like `StartsWith(char)` is faster than `IndexOf(char)`, especially for longer string

Reply:
Oh! Of course, in the failure case it would certainly be arbitrarily faster..